app/controllers/chat_controller.rb

```diff
@@ -331,15 +331,39 @@ def dismiss_retention_reminder
   def quote_messages
     params.require(:message_ids)
 
-    @chat_channel = ChatChannel.find_by(id: params[:chat_channel_id])
-    raise Discourse::NotFound if @chat_channel.blank?
-    raise Discourse::InvalidAccess if !guardian.can_see_chat_channel?(@chat_channel)
+    @chat_channel = find_channel_and_check_access(params[:chat_channel_id])
 
     message_ids = params[:message_ids].map(&:to_i)
     markdown = ChatTranscriptService.new(@chat_channel, messages_or_ids: message_ids).generate_markdown
     render json: success_json.merge(markdown: markdown)
   end
 
+  def move_messages_to_channel
+    params.require(:message_ids)
+    params.require(:destination_channel_id)
```

Comment:
Do we need to ensure `chat_channel_id` is present too?

Reply:
It's part of the route, so it should be there otherwise we won't hit this method :)